platform/jobs/edxPlatformPythonUnitTestsMaster.groovy

```diff
@@ -0,0 +1,153 @@
+    platformCredential : n/a
+    platformCloneReference : clone/.git
+    email : email-address@email.com
+    hipchat : token
+*/
+
+Map <String, String> predefinedPropsMap  = [:]
+predefinedPropsMap.put('GIT_SHA', '${GIT_COMMIT}')
+predefinedPropsMap.put('GITHUB_ORG', 'edx')
+predefinedPropsMap.put('GITHUB_REPO', 'edx-platform')
+predefinedPropsMap.put('TARGET_URL', JENKINS_PUBLIC_BASE_URL + 'job/edx-platform-python-unittests-master/${BUILD_NUMBER}/')
+predefinedPropsMap.put('CONTEXT', 'jenkins/python')
+
+/* stdout logger */
+/* use this instead of println, because you can pass it into closures or other scripts. */
+/* TODO: Move this into JenkinsPublicConstants, as it can be shared. */
+
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
+
+stringParam = [
+    name: 'sha1',
+    description: 'Sha1 hash of branch to build. Default branch : master',
+    default: 'refs/heads/master' 
+]
+
+/* Environment variable (set in Seeder job config) to reference a Jenkins secret file */
+String secretFileVariable = 'EDX_PLATFORM_TEST_PYTHON_SECRET'
+
+/* Map to hold the k:v pairs parsed from the secret file */
+Map secretMap = [:]
+try {
+    out.println('Parsing secret YAML file')
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    Thread thread = Thread.currentThread()
+    Build build = thread?.executable
+    Map envVarsMap = build.parent.builds[0].properties.get("envVars")
+    secretMap = JENKINS_PUBLIC_PARSE_SECRET.call(secretFileVariable, envVarsMap, out)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    /* TODO: Use/Build a more robust test framework for this */
+    assert jobConfig.containsKey('open')
+    assert jobConfig.containsKey('jobName')
+    assert jobConfig.containsKey('testengUrl')
+    assert jobConfig.containsKey('platformUrl')
+    assert jobConfig.containsKey('testengCredential')
+    assert jobConfig.containsKey('platformCredential')
+    assert jobConfig.containsKey('platformCloneReference')
+    assert jobConfig.containsKey('hipchat')
+    assert jobConfig.containsKey('email')
+    buildFlowJob(jobConfig['jobName']) {
+
+        /* For non-open jobs, enable project based security */
+        if (!jobConfig['open'].toBoolean()) {
+            authorization {
+                blocksInheritance(true)
+                permissionAll('edx')
+            }
+        }
+
+        parameters {
+            stringParam(stringParam.name, stringParam.default, stringParam.description)
+        }        
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR() //Discard build after 14 days
```

Comment:
nit: don't specify 14 days in the comment, because it could change back at JenkinsPublicConstants.